tests/unittests/core/test_experiment.py

```diff
@@ -626,14 +630,39 @@ def fake_handle(self, xxx):
         self.times_called += 1
         return self.times_called
 
-    def test_reserve_with_score(self, hacked_exp, exp_config):
+    def test_reserve_with_score(self, hacked_exp, exp_config, random_dt):
         """Reserve with a score object that can do its job."""
         self.times_called = 0
         hacked_exp.configure(exp_config[0][3])
         trial = hacked_exp.reserve_trial(score_handle=self.fake_handle)
         exp_config[1][6]['status'] = 'reserved'
+        exp_config[1][6]['heartbeat'] = random_dt
         assert trial.to_dict() == exp_config[1][6]
 
+    def test_fix_lost_trials(self, hacked_exp):
+        """Test that a running trial with an old heartbeat is set to interrupted."""
+        exp_query = {'experiment': hacked_exp.id}
+        trial = hacked_exp.fetch_trials(exp_query)[0]
+        heartbeat = datetime.datetime.strptime('2000-01-01T00:00:00', '%Y-%m-%dT%H:%M:%S')
+
+        Database().write('trials', {'status': 'reserved', 'heartbeat': heartbeat},
+                         {'experiment': hacked_exp.id, '_id': trial.id})
+
+        exp_query['status'] = {'$in': ['reserved']}
+        exp_query['_id'] = trial.id
+
+        assert hacked_exp.fetch_trials(exp_query)
+
+        hacked_exp.fix_lost_trials()
+
+        assert not hacked_exp.fetch_trials(exp_query)
+
+        exp_query.pop('status')
+
+        trial = hacked_exp.fetch_trials(exp_query)[0]
+
+        assert trial.status == 'interrupted'
+
```

Comment:
I know other tests could fail if it was not the case anyway, but we should add one test to explicitly test that the non-lost trials are not set to interrupted.